src/event.rs

```diff
@@ -158,3 +158,13 @@ pub enum EventType<TYPES: NodeType> {
         sender: TYPES::SignatureKey,
     },
 }
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub enum HotShotAction {
+    Vote,
+    Propose,
+    DAPropose,
+    DAVote,
+    DACert,
```

Comment:
Should this include `UpgradePropose` and `UpgradeVote`?

Reply:
I'm not sure, what do you think?  Will it be something we slash for if proposing multiple times in a view?  Or if we want to be sure not try to send an upgrade again after a restart it should be here

Reply:
I don't know for sure either, I was just thinking of it as being similar to DA proposals/votes. 

We probably do want to avoid re-proposing upgrades though, especially if we allow proposing further ahead of time.

Reply:
Makes sense to me, I'll add them both